### ISSUE TITLE
Replace Hyrule's `smacrolet` with a version based on our new `let`

### ISSUE THREAD

**Issue by Kodiologist** [hylang/hy#2192]

**Comment by allison-casey** [comment]
What does `smacrolet` do? The docs have no examples and their are no unit tests in hyrule.

**Comment by allison-casey** [comment]
Am i missing something with `smacrolet`? because this was really easy to implement, but i've never used `smacrolet` before so :woman_shrugging:. 

```clojure
(with-decorator
  (pattern-macro "smacrolet" [(brackets (many (+ FORM SYM)))
                              (many FORM)])
  (defn compile-smacrolet
    [compiler expr root bindings body]
    (setv res (Result)
          bindings (get bindings 0)
          scope (compiler.scope.create ScopeLet))
    (for [[target value] bindings]
      (scope.add target (str value)))
    (with [scope]
      (+ res (compiler.compile (.replace (mkexpr "do" #* body) expr))))))

(do (setv c 24)
    (smacrolet [b c]
      (setv foo (fn [a [b 1]]
                  (* b (inc a))))
      (* b (foo 7))))
;; => 192
```

```python
c = 24
foo = lambda a, b=1: b * inc(a)
c * foo(7)
# => 192
```

**Comment by Kodiologist** [comment]
I'm not familiar with it either, but hey, if it passes the tests, it passes the tests.

**Comment by allison-casey** [comment]
well this won't pass the tests because its a result macro and og `smacrolet`'s tests does a s-expression comparison with `macroexpand`. I'll rewrite them and see if they're in fact actually identical

**Comment by allison-casey** [comment]
also @Kodiologist is this supposed to go in `hyrule` or `hy`?

**Comment by Kodiologist** [comment]
I put this issue in Hy, but now that I think about it, it should go in Hyrule if you can do it as a regular macro instead of a result macro.

**Comment by allison-casey** [comment]
closed in hylang/hyrule#17
